### ISSUE TITLE
ExopPlayer 2.1.1 and 2.2.0 keep using network after player.release()

### ISSUE THREAD

**Issue by Yonezpt** [google/ExoPlayer#2425]
Is there any way to completely stop the ExoPlayer? Currently using `release()`, even coupled with `stop()`, does not stop the player from exercising network communications. This is what the console looks like after releasing the player, just that, there is no activity being closed here or any user action after releasing the player. It stays like this forever:

![ss](https://cloud.githubusercontent.com/assets/8498057/22651987/3bd94ae8-ec74-11e6-9412-63745abac067.png)

And nothing like this occurs before launching the ExoPlayer, so there is no reason to assume something else could be responsible for this besides the player.

**Comment by Yonezpt** [comment]
Forgot to include the Exoplayer version, it's 2.1.1 and the issue is still present in 2.2.0

**Comment by ojw28** [comment]
That definitely shouldn't happen, but I cannot reproduce the issue either. Please provide proper reproduction steps, preferably using the ExoPlayer demo app or some minor modification of it.

**Comment by Yonezpt** [comment]
I am very ashamed, there is no problem. When I started porting the ExoPlayer to my custom Activity I temporarily commented the state changes (`onResume` `onStop` `onPause` etc.) to see how the background playback was working. After doing that and pressing "Run demo" while still playing a video, the previous "player ghost" kept playing on the background of the new updated activity and so on, this would propragate until the app was closed or restarted. As a result there was almost at any time at least one video in the background consuming network resources, so when I released whichever was the most recent one, the old one kept running because it was no longer attached to that view because at that point it was essentially behaving as a background service, bound to no visible activity.

Lesson learned, from now on always use "Rerun" instead, sorry for the trouble, this problem was due to human error, the ExoPlayer is behaving correctly.
